### PR TITLE
Alphabetise team list in super admin

### PR DESCRIPTION
**Pull request by tommotaylor**
Alphabetising the list on the "show organisation page" in super admin. Must have been missed off the original task.

Involves a refactor using Clean Architecture to DRY up the code. Not 100% sure this is the right way to do it with CA so welcome feedback.

Screenshot 2019-04-01 at 11 38 03

### CODE REVIEW

File: spec/gateways/organisation_users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    it 'fetches the users for the organisation' do
+      expect(gateway.fetch).to eq(result)
+    end
+  end
+
+  context 'without users' do
+    let(:organisation) { create(:organisation) }
+
+    it 'returns an empty collection' do
+      expect(gateway.fetch).to eq([])
```
Comment (cybojenix) [inline spec/gateways/organisation_users_spec.rb]: We're expecting this gateway to always yield an active record collection. Can it be tested as such please?

I think we're also missing a test to assert that further queries can be run.
I'm unsure on the "rails way" for this, however I think asserting that it's an `ActiveRecord::Collection` should suffice

File: spec/use_cases/administrator/sort_users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+describe UseCases::Administrator::SortUsers do
```
Comment (cybojenix) [inline spec/use_cases/administrator/sort_users_spec.rb]: can there also be a test to ensure values fetched from the database get sorted?
This allows us to assert ordering outside of a feature test

Comment (tommotaylor) [reply spec/use_cases/administrator/sort_users_spec.rb]: I guess what you are talking about is the integration test that ties this all together. It currently exists in a few feature specs.

File: spec/gateways/organisation_users_spec.rb (hunk 1 above)
Comment (tommotaylor) [reply spec/gateways/organisation_users_spec.rb]: Agree to a point. Heres the behaviour:
```
gateway.fetch
=> []
```
```
gateway.fetch.class
=> User::ActiveRecord_Associations_CollectionProxy
```

File: spec/use_cases/administrator/sort_users_spec.rb (hunk 2 above)
Comment (tommotaylor) [reply spec/use_cases/administrator/sort_users_spec.rb]: Done @cybojenix

**Pull request merged**